src/frontends/lean/interactive.cpp

```diff
@@ -129,7 +129,7 @@ void report_info(environment const & env, options const & opts, io_state const &
             case break_at_pos_exception::token_context::import: {
                 auto parsed = parse_import(tk.to_string());
                 try {
-                    auto base_dir = dirname(m_mod_info.m_mod);
+                    auto base_dir = dirname(m_mod_info.m_id);
                     auto f = find_file(path, base_dir, parsed.first, string_to_name(parsed.second),
```

Comment:
What is being simplified?
BTW, I will soon need the following feature: load `.olean` files without loading proofs.

Reply:
Returning `module_info` from `load_module` instead of a tuple, mostly. It's mostly a preliminary refactoring for the next commit where a new module_info field is introduced.

Reply:
> I actually thought about implementing lazy loading of definition bodies some time ago, which for most proofs would mean they would never be loaded.

I have considered lazy loading too. I found one problem: if we don't eagerly load the file, are we going to keep it locked? If not, how do we make sure it is not modified? Do we create a copy? It seems to expensive. Then, I considered lazy deserialization. That is, we load the binary .olean eagerly, but deserialize on demand. I made same experiments, and it does save space, but, it is not a perfect solution. As we start to use more automation, the proofs will get much bigger, and they consume most of the space. I thought we could save a lot of space by just not loading them at all.